src/packer.ts

```diff
@@ -83,21 +83,68 @@ type PackState = {
   pointers: Pointer[];
 };
 
+export type PackArgs<Type extends readonly string[]> = {
+  /**
+   * The types of the values to pack.
+   */
+  types: Type;
+
+  /**
+   * The values to pack.
+   */
+  values: TypeMap<Type, 'input'>;
+
+  /**
+   * Whether to use the non-standard packed mode.
+   */
+  packed?: boolean | undefined;
+
+  /**
+   * Whether to use tight packing mode. Only applicable when `packed` is true.
+   * When true, the packed mode will not add any padding bytes. This matches
+   * the packing behaviour of `ethereumjs-abi`, but is not standard.
+   */
+  tight?: boolean | undefined;
+
+  /**
+   * Whether to use the non-standard packed mode in "array" mode. This is
+   * normally only used by the {@link array} parser.
+   */
+  arrayPacked?: boolean | undefined;
```

Comment:
I wonder if this should be exposed to the user as it is sort of an implementation detail, what do you think?

Reply:
It's not, since `packer.ts` isn't exported directly. You have to call it through `encodePacked`, which does not have an option for this.

Reply:
Ah, great!